Modify info api endpoint to return last block before the update of checksum

QYuQianchen opened the pull request:
Add a new field in `get_last_indexed_block` and update the return structure of `info` endpoint
Closes https://github.com/hoprnet/hoprnet/issues/6440

Sample return:
```json
{
  "announcedAddress": [
    "/ip4/127.0.0.1/tcp/19091"
  ],
  "chain": "anvil",
  "channelClosurePeriod": 15,
  "connectivityStatus": "Green",
  "hoprChannels": "0x9a9f2ccfde556a7e9ff0848998aa4a0cfd8863ae",
  "hoprManagementModule": "0xa51c1fc2f0d1a1b8494ed1fe312d7c3a78ed91c0",
  "hoprNetworkRegistry": "0x3aa5ebb10dc797cac828524e59a333d0a371443c",
  "hoprNodeSafe": "0x6a64fe01c3aba5bdcd04b81fef375369ca47326f",
  "hoprNodeSafeRegistry": "0x0dcd1bf9a1b36ce34237eeafef220932846bcd82",
  "hoprToken": "0x9a676e781a523b5d0c0e43731313a708cb607508",
  "indexBlockPrevChecksum": 70,
  "indexerBlock": 126,
  "indexerChecksum": "0x91bd8217ccfac3a51eeb1e7f15d3af716162638f48717df305503a061558a13c",
  "isEligible": true,
  "listeningAddress": [
    "/ip4/127.0.0.1/tcp/19091"
  ],
  "network": "anvil-localhost"
}
```

NumberFour8 reviewed (dismissed):
An existing migration cannot be modified like this. A new migration needs to be added, which alters the existing table - adding the new column with a reasonable default value, so that existing databases are migrated correctly to the new schema.

QYuQianchen commented:
> An existing migration cannot be modified like this. A new migration needs to be added, which alters the existing table - adding the new column with a reasonable default value, so that existing databases are migrated correctly to the new schema.

My bad! Reverted those changes and added new migration logic

Teebor-Choka commented on chain/indexer/src/block.rs:
```diff
@@ -97,7 +97,7 @@ where
         let tx_significant_events = self.egress.clone();
 
-        let (db_latest_block, checksum) = self.db.get_last_indexed_block(None).await?;
+        let (db_latest_block, checksum, _) = self.db.get_last_indexed_block(None).await?;
```
At this point it'd be cleaner to replace the tuple with a dedicated struct with public members.

Teebor-Choka commented on db/api/src/info.rs:
```diff
@@ -115,7 +115,7 @@ pub trait HoprDbInfoOperations {
 
     /// Retrieves the last indexed block number.
-    async fn get_last_indexed_block<'a>(&'a self, tx: OptTx<'a>) -> Result<(u32, Hash)>;
+    async fn get_last_indexed_block<'a>(&'a self, tx: OptTx<'a>) -> Result<(u32, Hash, u32)>;
```
```suggestion
    pub struct DescribedBlock {
        current: u32,
        checksum: Hash
        previous: u32
    }

    async fn get_last_indexed_block<'a>(&'a self, tx: OptTx<'a>) -> Result<DescribedBlock>;
```

Teebor-Choka commented on db/api/src/info.rs:
```diff
@@ -437,7 +442,9 @@ impl HoprDbInfoOperations for HoprDb {
-                        debug!("updating block checksum {current_checksum} -> {new_hash} @ {block_num}");
+                        // when a new checksum is computed, we need to update previous_indexed_block_prio_to_checksum_update
+                        active_model.previous_indexed_block_prio_to_checksum_update = Set(current_last_indexed_block);
+                        debug!("updating block checksum {current_checksum} @ {current_last_indexed_block} -> {new_hash} @ {block_num}");
```
```suggestion
                        trace!("updating block checksum {current_checksum} @ {current_last_indexed_block} -> {new_hash} @ {block_num}");
```

Teebor-Choka commented on db/api/src/peers.rs:
```diff
@@ -152,7 +152,7 @@ pub trait HoprDbPeersOperations {
     async fn get_network_peer(&self, peer: &PeerId) -> Result<Option<PeerStatus>>;
 
-    /// Returns a stream of all stored peers, optionally matching the given [SimpleExpr] filter.
+    /// Returns a stream of all stored peers, optionally matching the given SimpleExpr filter.
```
```suggestion
    /// Returns a stream of all stored peers, optionally matching the given [`SimpleExpr`] filter.
```

Teebor-Choka commented on db/migration/src/m20240810_000013_index_extend_chain_info_with_pre_checksum_block.rs:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        // manager.drop_table(Table::drop().table(Post::Table).to_owned()).await
```
```suggestion
```

Teebor-Choka reviewed (commented):
Minor changes.

QYuQianchen commented on scripts/utils.sh:
```diff
@@ -1,12 +1,27 @@
 #!/usr/bin/env bash
 
-# prevent execution of this script, only allow execution
+# prevent execution of this script, only allow sourcing
```
Applying this change would result in failures in integration tests. Thus, skip it for the moment in favor of a thorough clean up of all the bash scripts later
